gunpowder/nodes/generic_train.py

```diff
@@ -117,7 +117,11 @@ def setup(self):
     def prepare(self, request):
         deps = BatchRequest()
         for key in self.inputs.values():
-            deps[key] = request[key]
+            if key in self.array_specs:
+                deps[key] = self.array_specs[key].copy()
```

Comment:
I think we still want to add the `roi` from the request to our dependencies in this case, and only keep the voxel_size, dtype, etc. from `self.array_specs`